tools/vfat-resize.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2020  Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Vojtech Trefny <vtrefny@redhat.com>
+ *
+ * This is just a small helper program for (V)FAT filesystem resizing using libparted.
+ * We don't want this to be part of the library because libparted is licensed under GPLv3
+ * and we want to keep libblockdev LGPL compatible.
+ *
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <glib.h>
+#include <parted/parted.h>
+#include <bytesize/bs_size.h>
+
+
+void print_usage (const char *cmd) {
+    g_print ("Usage: %s device [size]\n"
+             "-h    --help   Print this usage info\n"
+             "If size is not specified the file system is adapted to the underlying block device\n",
+             cmd);
+}
+
+PedExceptionOption bd_exc_handler (PedException *ex) {
+    if (ex->type <= PED_EXCEPTION_WARNING && (ex->options & PED_EXCEPTION_IGNORE) != 0) {
+      g_print ("[parted] %s\n", ex->message);
+      return PED_EXCEPTION_IGNORE;
+    }
+    g_printerr ("[parted] %s\n",ex->message);
+    return PED_EXCEPTION_UNHANDLED;
+}
+
+int main (int argc, char *argv[]) {
+    PedDevice *ped_dev = NULL;
+    PedGeometry geom = {0};
+    PedGeometry new_geom = {0};
+    PedFileSystem *fs = NULL;
+    PedSector start = 0;
+    PedSector length = 0;
+    gint status = 0;
+    guint64 new_size = 0;
+    BSSize bs_size;
+    BSError *bs_error = NULL;
+    gchar *device = NULL;
+
+    if ((g_strcmp0 (argv[1], "-h") == 0) || g_strcmp0 (argv[1], "--help") == 0) {
```

Comment:
Huh, in case `argc == 0` accessing `argv[1]` may lead to segfault unless there's a NULL. Still, better to check `argc` first. Also there's an extra parenthesis (cosmetic issue).

Reply:
I think `argc` will always be at least 1 and `argv` is guaranteed to be NULL terminated so this couldn't happen, but I fixed it anyway to be safe.